common/cache.go

```diff
@@ -62,14 +62,13 @@ func (c *BlockCache) Add(height int, block *walletrpc.CompactBlock) (error, bool
 	// Invariant: m[firstBlock..nextBlock) are valid.
 
 	// Detect reorg, ingestor needs to handle it
-	if height > c.firstBlock && !bytes.Equal(block.PrevHash, c.m[height-1].hash) {
+	if block != nil && height > c.firstBlock && !bytes.Equal(block.PrevHash, c.m[height-1].hash) {
```

Comment:
The new `nil` check allows the unit test to force an error below, on the call to `proto.Marshal()`.

Reply:
It's worth adding a comment in the code to explain this.